cetmix_tower_server/ssh/ssh.py

```diff
@@ -0,0 +1,296 @@
+        self.timeout = timeout
+        self._ssh_client: Optional[SSHClient] = None
+
+    def connect(self) -> SSHClient:
+        """
+        Connect to the SSH server.
+        """
+        if self._ssh_client is not None:
+            return self._ssh_client
+
+        self._ssh_client = SSHClient()
+        self._ssh_client.load_system_host_keys()
+        self._ssh_client.set_missing_host_key_policy(AutoAddPolicy())  # type: ignore
+
+        connect_params = {
+            "hostname": self.host,
+            "port": self.port,
+            "username": self.username,
+            "allow_agent": self.allow_agent,
+            "timeout": self.timeout,
+        }
+
+        if self.mode == "p":
+            if not self.password:
+                raise ValueError("For password mode, you need to pass a password.")
+            connect_params["password"] = self.password
+        elif self.mode == "k":
+            if not self.ssh_key:
+                raise ValueError("For key mode, you need to pass an SSH key.")
+            connect_params["pkey"] = KeyLoader.load_private_key(self.ssh_key)
+        else:
+            raise ValueError(f"Unsupported connection mode: {self.mode}")
+
+        self._ssh_client.connect(**connect_params)
+        return self._ssh_client
+
+    def disconnect(self) -> None:
+        """
+        Disconnect the SSH connection.
+        """
+        if self._ssh_client:
+            _logger.info("SSHConnection: closing SSH connection.")
+            self._ssh_client.close()
+            self._ssh_client = None
+
+    def get_transport(self):
+        """
+        Get the SSH transport.
+        """
+        if self._ssh_client is None:
+            self.connect()
+        return self._ssh_client.get_transport()
+
+
+class SftpService:
+    """
+    Service for working with SFTP, using SSH connection.
+    """
+
+    def __init__(self, connection: SSHConnection):
+        """
+        Initialize the SftpService instance.
+        """
+        self.connection = connection
+        self._sftp_client: Optional[SFTPClient] = None
+
+    def get_client(self) -> SFTPClient:
+        """
+        Get the SFTP client.
+        """
+        if self._sftp_client is None:
+            transport = self.connection.get_transport()
+            self._sftp_client = SFTPClient.from_transport(transport)
+        return self._sftp_client
+
+    def upload_file(self, file: Union[str, io.BytesIO], remote_path: str) -> None:
+        """
+        Upload a file to the remote server.
+        """
+        client = self.get_client()
+        if isinstance(file, io.BytesIO):
+            client.putfo(file, remote_path)
+        elif isinstance(file, str):
+            client.put(file, remote_path)
+        else:
+            raise TypeError(f"File type {type(file).__name__} is not supported.")
+
+    def download_file(self, remote_path: str) -> bytes:
+        """
+        Download a file from the remote server.
```

Comment:
Can we add an option to download using sudo?

example: /etc/* usualy required sudo password to fetch the file.. .

Curently i use workaround: command with sudo to temporary copy file to location reachable without sudo...

Reply:
@badbole hm, indeed. I will add this to backlog and see what can be done here.

Reply:
@badbole check this once again. Paramiko doesn't allow to do it. So there should be some workaround developed for that. Which does actually the same thing as you are doing now. Let's put it into the backlog.